intro-to-R-tidyverse/04c-intro_to_tidyverse_exercise-part-2.Rmd

```diff
@@ -0,0 +1,170 @@
+
+### Set Up 
+
+In this notebook, we will be using tidyverse packages. 
+You can load the `library()` or use the `::` strategy. 
+
+```{r}
+
+```
+
+It's convenient to declare our directory names at the beginning so we can change
+it once and have it change throughout the notebook. 
+Remember we need to be consistent with what we used in the previous notebook. 
+
+```{r}
+data_dir <- 
+plots_dir <- 
+```
+
+## Read in the cleaned metadata
+
+Use `readr::read_tsv()` to read in the filtered gene expression matrix TSV, 
+"filtered_GSE19578.tsv" and assign it the name `genes_df`. 
+
+```{r}
+
+```
+
+Use `readr::read_tsv()` to read in the cleaned metadata TSV, "GSE19578.tsv" 
+and assign it the name `metadata`. 
+
+```{r}
+
+```
+
+## Set up grouped summary statistics for genes
+
+Let's calculate some fold changes for our groups. 
+We've set up a data frame that is transposed so you can use using `tidyverse` functions. 
+
+```{r}
+transposed_genes_df <- genes_df %>% 
+  # We want the gene information to be stored as rownames for the next set of 
+  # steps we are doing
+  tibble::column_to_rownames("Gene") %>% 
+  # Transposing the data frame so it is sample x genes
+  t() %>% 
+  # t() function turns our data frame into a matrix so let's turn it back to 
+  # a data.frame so we can use it with tidyverse functions
+  as.data.frame() %>% 
+  # By default our sample ids will be stored as rownames, let's bring it out as 
+  # its own columns 
+  tibble::rownames_to_column("sample_id")
+```
+
+Use `dplyr::select()` and an `%in%` statement to choose ~2-5 genes of interest. 
+Can use [GeneCard](https://www.genecards.org/) to find the corresponding Ensembl
+ID to your favorite genes. 
+If you don't have genes in particular you are interested in, feel free to use 
+these: `ENSG00000134323` (MYCN), `ENSG00000124762`(CDKN1A), and 
+`ENSG00000115414` (FN1).
+
+```{r}
+genes_oi <- c("ENSG00000134323", 
+              "ENSG00000124762", 
+              "ENSG00000115414")
+
+```
+
+Use `dplyr::inner_join()` to join the `metadata_clean` columns to 
+`transposed_genes_df` by the `sample_id`. 
+
+```{r}
+
+```
+
+Calculate the group means of your `select()`ed genes. 
+
+```{r}
+
+```
+
+### Plot the data
+
+Now that we have our data set up, we will make some plots!
+
+**Plot 1) Scatterplot**
+
+Set up a ggplot scatterplot with either one or two of the selected genes in
+`genes_of_interest_df` and/or with `age`. 
+
+```{r} 
+
+```
+
+**Plot 2) Facet wrapped jitter**
+
+Create a jitter plot of each gene where each `disease` group has its own plot 
+using `facet_wrap()`. 
+
+We've done the first step for you. 
+Your data will need to be in a long format. 
+Explore the `long_form_genes_oi_df` data frame to see how we've transformed it 
+with `tidyr::gather()`. 
+
+```{r}
+# Reformat to long format data frame 
+long_form_genes_oi_df <- genes_of_interest_df %>% 
```

Comment:
If we are going to use the `genes_of_interest_df` object here, we need to tell them to specifically define this object above when creating their genes of interest data frame.

Reply:
So, I'm debating on how prescriptive to be in this notebook since it is the third one. The idea is we want them to try things and there are many ways that some of these steps could be done. IDK.

Reply:
Ah but I see what you are saying. Because this step is already done.

Reply:
I've added a more explicitly prompt.